x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -75,45 +96,36 @@ export const RulesTable = ({
       isSelectable={true}
       selection={selection}
       itemId={(v) => v.id}
+      rowProps={rowProps}
     />
   );
 };
 
-const ruleNameRenderer = (name: string) => (
-  <EuiLink className="eui-textTruncate" title={name}>
-    {name}
-  </EuiLink>
-);
-
-const timestampRenderer = (timestamp: string) =>
-  moment.duration(moment().diff(timestamp)).humanize();
-
-interface GetColumnProps {
+interface GetColumnProps extends Pick<RulesTableProps, 'setSelectedRuleId'> {
   toggleRule: (rule: RuleSavedObject) => void;
 }
 
-const createRuleEnabledSwitchRenderer =
-  ({ toggleRule }: GetColumnProps) =>
-  (value: boolean, rule: RuleSavedObject) =>
-    (
-      <EuiSwitch
-        showLabel={false}
-        label={value ? TEXT.DISABLE : TEXT.ENABLE}
-        checked={value}
-        onChange={() => toggleRule(rule)}
-        data-test-subj={TEST_SUBJECTS.getCspRulesTableItemSwitchTestId(rule.attributes.id)}
-      />
-    );
-
 const getColumns = ({
   toggleRule,
+  setSelectedRuleId,
 }: GetColumnProps): Array<EuiTableFieldDataColumnType<RuleSavedObject>> => [
   {
     field: 'attributes.name',
     name: TEXT.RULE_NAME,
     width: '60%',
     truncateText: true,
-    render: ruleNameRenderer,
+    render: (name, rule) => (
+      <EuiLink
```

Comment:
nit: you can use `EuiButtonEmpty` for the same UI since you don't actually use this as a link. not critical at all just mentioning it in case you missed that option.